[Bug Fix] Fix CUDA EmbeddingBag to use correct offset value when include_last_offset is set

chirayuG-nvidia opened the pull request:
Bug fix to compute correct offset location for last embedding bag when `include_last_offset` is set.

Fixes #76885

swolchok committed:
[PyTorch] Don't enter MHA fast path when bias & query dtypes don't match

Pull Request resolved: https://github.com/pytorch/pytorch/pull/76879

The fast path does not support this: transform_bias_rescale_qkv will try to grab bias.data_ptr() assuming the dtypes are the same. (Also, I have no idea how this happens.)

Differential Revision: [D36156872](https://our.internmc.facebook.com/intern/diff/D36156872/)

Approved by: https://github.com/cpuhrsch

swolchok committed:
[PyTorch] Add corner case tests for IValue::use_count()

Pull Request resolved: https://github.com/pytorch/pytorch/pull/76418

There's some weird behavior currently with this accessor. Capture it in a test.

Differential Revision: [D31154949](https://our.internmc.facebook.com/intern/diff/D31154949/)

Approved by: https://github.com/ezyang

swolchok committed:
[PyTorch] Replace IValue::intrusive_ptr with IValue::isIntrusivePtr()

Pull Request resolved: https://github.com/pytorch/pytorch/pull/76425

This tidies up an extra piece of state that was computable from the other state in IValue.

Differential Revision: [D31290017](https://our.internmc.facebook.com/intern/diff/D31290017/)

Approved by: https://github.com/ezyang

ezyang committed:
Support Tensor source for x.set_(storage, offset, size, strides)

This is convenient for cases where we don't have Storage bound
correctly (e.g., meta tensors).  It is also consistent with a universe
where we get rid of storages, although arguably this is never
gonna happen.

Signed-off-by: Edward Z. Yang <ezyangfb.com>

Pull Request resolved: https://github.com/pytorch/pytorch/pull/77007

Approved by: https://github.com/ngimel

chirayuG-nvidia committed:
Fix CUDA EmbeddingBag to use correct offset value when include_last_offset is set

chirayuG-nvidia committed:
Fix CUDA EmbeddingBag to use correct offset value when include_last_offset is set

ezyang committed:
Add OpInfo based meta tensor tests [RELAND]

PR #75994 was taking too long to ship so I extracted out the CrossRef gadget and
had it run on a simple OpInfo invocation only.

Signed-off-by: Edward Z. Yang <ezyangfb.com>

Pull Request resolved: https://github.com/pytorch/pytorch/pull/77008

Approved by: https://github.com/ngimel

Commit:
Adds decorators for Python References and extends Python Reference testing (#76945)

This PR does the following...

Tests:
- fixes test_type_promotion in test_binary_ufuncs to correctly generate scalar cpu tensors
- fixes test_python_reference_consistency to use the Python Reference's reference inputs
- extends Python reference testing to test_conj_view, test_neg_view, and test_neg_conj_view
- adds a NaN propagation sample input for elementwise unary and binary operations
- fixes the UnaryUfuncInfo class to properly register its reference inputs
- Updates the Python Reference OpInfos to skip error inputs when their behavior on scalar inputs is inconsistent with their reference operators

Code organization:
- moves elementwise type promotion functionality to prims.utils

Prims & Refs:
- fixes scalar cpu tensor handling by having them pass through broadcasting and device and shape checks
- adds two decorators, `elementwise_type_promotion_wrapper` and `out_wrapper`, the former allows for elementwise type promotion to be automated and the latter automatically adds the out kwarg and handles it properly

cc @ezyang who also had some thoughts on cpu scalar tensor handling
cc @chillee -- might want to use this new decorator as we converge decompositions and references
Pull Request resolved: https://github.com/pytorch/pytorch/pull/76945
Approved by: https://github.com/ngimel

chengjunlu committed:
Fix issue in s_addmm_out_sparse_dense_cpu only supporting CUDA device checking (#77018)

## Motivation
The at::native::s_addmm_out_sparse_dense_cpu only supports the CPU tensors. But it only checks whether the tensor is on CUDA device which is not enough.

## Solution
Change the tensor device type checkging from is_cuda to !is_cpu to protect other backends than the CUDA.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/77018
Approved by: https://github.com/IvanYashchuk

ysiraichi committed:
Move `contiguous_strides` to `c10/util/strides.h`.

This PR moves `contiguous_strides` implementation from `LinearAlgebraUtils.h` to a broader
scope: `c10/util/strides.h`. This should make it easier to reuse this implementation on
other parts of the codebase.

Pull Request resolved: https://github.com/pytorch/pytorch/pull/76812

Approved by: https://github.com/ezyang

zasdfgbnm committed:
[Reland] Add atan2 isfinite isinf isnan isneginf isposinf isreal to nvfuser and its frontend (#76769)

This reverts commit 4bb59441339074515a4257704acca6ebd80826cc.

Pull Request resolved: https://github.com/pytorch/pytorch/pull/76769
Approved by: https://github.com/csarofeen, https://github.com/mruberry

Commit:
xfails bool add nvfuser test (#77031)

Per title, the test doesn't handle boolean alpha properly
Pull Request resolved: https://github.com/pytorch/pytorch/pull/77031
Approved by: https://github.com/ezyang

pytorchmergebot committed:
Revert "Move `contiguous_strides` to `c10/util/strides.h`."

This reverts commit 84316cc389c72cda6272e640039745e02bc33e8f.

Reverted https://github.com/pytorch/pytorch/pull/76812 on behalf of https://github.com/ezyang

mikaylagawarecki committed:
Add failure on torch.use_deterministic_agorithms(True) for scatter_reduce

Pull Request resolved: https://github.com/pytorch/pytorch/pull/77041

Approved by: https://github.com/cpuhrsch

ysiraichi committed:
Move `contiguous_strides` to `c10/util/strides.h`.

This PR moves `contiguous_strides` implementation from `LinearAlgebraUtils.h` to a broader
scope: `c10/util/strides.h`. This should make it easier to reuse this implementation on
other parts of the codebase.

Pull Request resolved: https://github.com/pytorch/pytorch/pull/76812

Approved by: https://github.com/ezyang

BioGeek committed:
Typo fix in generated module name (#76880)

`f"{_FILE_PREFIX}non_sriptable"` -> `f"{_FILE_PREFIX}non_scriptable"`

Pull Request resolved: https://github.com/pytorch/pytorch/pull/76880
Approved by: https://github.com/mrshenli

soulitzer commented:
Thanks for the fix! You might want to fix lint and rebase, the other failures don't look related

Commit:
type promote clamp (#77035)

Fixes #76630
When clamp(Tensor, Tensor) is structured, big parts of this PR won't be needed, but for now let's fix type promotion to make behavior more regular.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/77035
Approved by: https://github.com/mruberry

Commit:
use opmath and expm1 for elu (#77062)

Fixes #77054

Pull Request resolved: https://github.com/pytorch/pytorch/pull/77062
Approved by: https://github.com/mruberry

Commit:
[primTorch] flatten, squeeze, unsqueeze... (#77043)

This PR ...

Makes the following testing changes:

- Updates stride testing in test_python_reference_consistency to only check strides of dimensions with length > 1
- Creates reference inputs for reshape
- Creates reference inputs for chunk
- Extends the sample inputs for unsqueeze
- Extends the sample inputs for stack -- test_conj_view and test_neg_view are now xfailed
  - https://github.com/pytorch/pytorch/issues/77046

Makes the following architecture changes:
- Adds the refs.special (sub)module
- Adds the refs.nn.functional (sub)module

Adds the following prims:
- expand_dims
- view_of
- rev
- clone

Adds the following references:
  -  flatten
  - squeeze
  - unsqueeze
  - special.i0e
  - special.i1e
  - logical_or
  - logical_and
  - isclose
  - flip
  - stack
  - nn.functional.elu
  - chunk
  - clone
  - narrow

Identifies the following bugs in PyTorch today:
- https://github.com/pytorch/pytorch/issues/77054
- https://github.com/pytorch/pytorch/issues/77055

Pull Request resolved: https://github.com/pytorch/pytorch/pull/77043
Approved by: https://github.com/ngimel

IvanYashchuk committed:
ldl_factor fix magma version check (#77019)

The tests were failing for CUDA 11.6 environment because it is using MAGMA 2.6.1 and preprocessor macros were incorrectly checking the version.

Fixes #76961

Pull Request resolved: https://github.com/pytorch/pytorch/pull/77019
Approved by: https://github.com/Lezcano, https://github.com/mruberry

Commit:
updates and encodes clamp xfails (#77077)

This should resolve clamp CI test failures that aren't using disabled issues
Pull Request resolved: https://github.com/pytorch/pytorch/pull/77077
Approved by: https://github.com/janeyx99

janeyx99 committed:
Manually skip test_sparse_addmm as disable code is not working for now (#77076)

Related to https://github.com/pytorch/pytorch/issues/73145

It was previously skipped for Linux and Windows, but mac has become a problem as well.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/77076
Approved by: https://github.com/ezyang

janeyx99 committed:
[disable bot] Handle empty bodies (#77078)

Fixes problem where the disable bot just breaks when the issue body is empty. Disable bot started failing when #77020 was introduced as it had no issue description.

This PR simply adds a condition.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/77078
Approved by: https://github.com/ezyang, https://github.com/malfet

desertfire committed:
[LT] Move device lock in LazyGraphExecutor to a later place

Summary: cherry-picking https://github.com/pytorch/xla/pull/3457

Pull Request resolved: https://github.com/pytorch/pytorch/pull/76974

Approved by: https://github.com/wconstab, https://github.com/JackCaoG

leslie-fang-intel committed:
New generated conv_bn folding should use same weight and bias dtype as original conv module (#77042)

When doing the conv_bn folding in `torch.jit.freeze`, the new calculated `weight` and `bias` for the new conv will be promoted to high precision such as `float32` even the original `weight` and `bias` for the conv is low precision such as `bfloat16`.
In this PR, we will record the original dtype for the conv's `weight` and `bias` and convert the data type back after `conv_bn` folding.

Pull Request resolved: https://github.com/pytorch/pytorch/pull/77042
Approved by: https://github.com/eellison

simonJJJ committed:
fix linalg inv doc (#77079)

I believe it is a doc typo, so I fix it.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/77079
Approved by: https://github.com/Lezcano

Commit:
shard `win-vs2019-cuda11.3-py3 / test` from 2 shards to 5 shards (#76867)

Fixes #ISSUE_NUMBER
shard `win-vs2019-cuda11.3-py3 / test` from 2 shards to 5 shards
helps w/ #76838

Notes:
- avg tts for the past week as of May 5 is 4.7 and 4.5 hours for 1st and 2nd shard on master, around 4 hours for all branches (but I don't think the changes from removing distributed tests + moving testing off of pull have come into effect yet)
- high overhead
- hope that tts doesn't explode

Sharding spreadsheet: https://docs.google.com/spreadsheets/d/1BdtVsjRr0Is9LXMNilR02FEdPXNq7zEWl8AmR3ArsLQ/edit#gid=1153012347

Pull Request resolved: https://github.com/pytorch/pytorch/pull/76867
Approved by: https://github.com/suo, https://github.com/janeyx99

jjsjann123 committed:
Torchvision patch (#77001)

Fixes #76791

Note that this is a hot patch so we get to run upstream tests. I'm doing proper fix in our local repo and will update upstream code once those are merged/reviewed.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/77001
Approved by: https://github.com/davidberard98

fuqianz committed:
[Bootcamp]Add option for flatbuffer loader to copy memory to individual tensors (#76986)

Summary: Add option for flatbuffer loader to copy memory to individual tensors to allow free memeory without waiting for all tensor runs completed.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/76986
Approved by: https://github.com/qihqi

Krovatkin committed:
LT tutorial [WIP] (#76392)

Fixes #ISSUE_NUMBER

Pull Request resolved: https://github.com/pytorch/pytorch/pull/76392
Approved by: https://github.com/wconstab

Krovatkin committed:
Revert "Revert "[WIP] customize the C++ class for valueT"" (#77003)

This reverts commit ec841b0346ade6664d13d5d0263b8e6990bf4d95.

Fixes #ISSUE_NUMBER

Pull Request resolved: https://github.com/pytorch/pytorch/pull/77003
Approved by: https://github.com/shunting314, https://github.com/JackCaoG

chirayuG-nvidia committed:
remove tabs from embedding fix

chirayuG-nvidia committed:
Merge branch 'embeddingBagFix' of https://github.com/chirayuG-nvidia/pytorch into embeddingBagFix

ngimel commented:
Please rebase and reopen, this can't be multi-K line PR.
